inc/ThirdParty/Plugins/ModPagespeed.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Plugins;
+
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Subscriber for compatibility with Mod_pagespeed
+ *
+ * @since  3.8.2
+ */
+class ModPagespeed implements Subscriber_Interface {
+
+	/**
+	 * Subscribed events for Pagespeed module.
+	 *
+	 * @since  3.8.2
+	 * @inheritDoc
+	 */
+	public static function get_subscribed_events(): array {
+		return [
+			'admin_notices' => 'show_admin_notice',
+		];
+	}
+
+	/**
+	 * Check if mod_pagespeed is enabled on this server with and cache the result on transient.
+	 *
+	 * @since  3.8.2
+	 *
+	 * @return bool Status of mod_pagespeed.
+	 */
+	private function has_pagespeed_with_cache() {
+		$has_pagespeed = get_transient( 'rocket_mod_pagespeed_enabled' );
+
+		if ( false === $has_pagespeed ) {
+			$has_pagespeed = self::has_pagespeed();
+			set_transient( 'rocket_mod_pagespeed_enabled', (int) $has_pagespeed, DAY_IN_SECONDS );
+		}
+
+		return 1 === (int) $has_pagespeed;
+	}
+
+	/**
+	 * Check if mod_pagespeed is enabled on this server.
+	 *
+	 * @since  3.8.2
+	 *
+	 * @return bool
+	 */
+	private static function has_pagespeed(): bool {
+		global $is_apache;
+
+		if ( $is_apache && function_exists( 'apache_get_modules' ) ) {
+			if ( in_array( 'mod_pagespeed', apache_get_modules(), true ) ) {
+				return true;
+			}
+		}
+
+		$home_request = wp_remote_get(
+			home_url(),
+			[
+				'timeout'   => 0.01,
```

Comment:
that's a very low timeout, what if we don't receive the response in that timing?

Reply:
Ok I will leave it as default (5 seconds)